src/names/name-resolution.md

```diff
@@ -1,4 +1,384 @@
+>
+> ```rust
+> mod bar {
+>     pub const NAME: bool = true;
+> }
+>
+> mod baz {
+>     pub const NAME: bool = false;
+> }
+>
+> use baz::NAME;
+>
+> pub fn foo() {
+>     use bar::*;
+>     assert!(NAME);
+>     //      ^--- This `NAME` is resolved during primary resolution.
+> }
+> ```
+
+r[names.resolution.expansion.imports.ambiguity.moreexpandedvsouter]
+Name bindings from macro expansions to may not shadow name bindings from outside of those expansions.
+
+```rust,compile_fail,E0659
+macro_rules! name {
+    () => {}
+}
+
+macro_rules! define_name {
+    () => {
+        macro_rules! name {
+            () => {}
+        }
+    }
+}
+
+fn foo() {
+    define_name!();
+    name!(); // ERROR: `name` is ambiguous
+}
+```
+
+r[names.resolution.expansion.imports.ambiguity.pathvstextualmacro]
+Path-based scope bindings for macros may not shadow textual scope bindings to macros. For bindings from [`use` declarations], this applies regardless of their [sub-namespace].
+
+```rust,compile_fail,E0659
+#[macro_export]
+macro_rules! m2 {
+    () => {}
+}
+macro_rules! m {
+    () => {}
+}
+pub fn foo() {
+    m!(); // ERROR: `m` is ambiguous
+    use crate::m2 as m; // In scope for entire function body.
+}
+```
+
+r[names.resolution.expansion.imports.ambiguity.builtin-attr]
+It is an error to use a user defined attribute or derive macro with the same name as a builtin attribute (e.g. inline).
+
+<!-- ignore: test doesn't support proc-macro -->
+```rust,ignore
+// myinline/src/lib.rs
+use proc_macro::TokenStream;
+
+#[proc_macro_attribute]
+pub fn inline(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    item
+}
+```
+
+<!-- ignore: requires external crates -->
+```rust,ignore
+// src/lib.rs
+use myinline::inline;
+use myinline::inline as myinline;
+
+#[myinline::inline]
+pub fn foo() {}
+
+#[crate::inline]
+pub fn bar() {}
+
+#[myinline]
+pub fn baz() {}
+
+#[inline] // ERROR: `inline` is ambiguous
+pub fn qux() {}
+```
+
+r[names.resolution.expansion.macros]
+### Macros
+
+r[names.resolution.expansion.macros.intro]
+Macros are resolved by iterating through the available scopes to find the available candidates. Macros are split into two sub-namespaces, one for bang macros, and the other for attributes and derives. Resolution candidates from the incorrect sub-namespace are ignored.
+
+r[names.resolution.expansion.macros.visitation-order]
+The available scopes are visited in the following order.
+
+* [derive helpers]
+* [textual scope macros]
+* [path-based scope macros]
+* [`macro_use` prelude]
+* [Standard library prelude]
+* [Builtin attributes]
+
+> [!NOTE]
+>
+> The compiler will attempt to resolve derive helpers that are used before
```

Comment:
Unwrap.

(I realize blockquotes don't look terribly great in source. I'm wondering if we should just choose a different syntax...)